Tolerant base64 decoding.  Accept url-safe or standard encoding in responses.

dantreiman opened the pull request:
Attempts to decode `bytes` fields in response JSON using both `standard` and `url_safe` encodings.  The expected encoding's desierializer is tried first.  On error the second is attempted.  If the second fails, the original error from the first is returned.

Fixes https://github.com/Byron/google-apis-rs/issues/502

Likely fixes other bugs too but I haven't tested

Byron reviewed (changes requested):
That's great, I am so looking forward to getting this released.

Could you take a look at the failing tests and adjust them? You could even add some assertions to assure both serializers work as expected (and that one failure already indicates it does as it doesn't fail anymore).

Thanks again!

dantreiman commented:
Sounds good.  I've added two cases using `+`, `/` and `-`, `_` respectively to ensure decoder doesn't error out.  Changed failure case to a character outside either accepted base64 character set.

Byron reviewed (approved):
That's great, thanks so much!
